### PR TITLE
WIP: Dunify

### PR DESCRIPTION
**Pull request by mseri**
This is a work in progress, I have re-picked it up because I believe that vendoring openblas in owl is the way to go, and this provides the necessary test bed.

The latest version seems to work fine in executables and utop, but it is still too artificial.

Current issues (feel free to add more if you spot them):

- [ ] building of `libeigen_cpp_stubs.a` happens too late and the ordering has to be enforced by hand
- [x] ~~the cpp library is not linked at the right time and I had to add it by hand in the `eigen/dune` `c_library_flags`~~
- [ ] some flags are probably wrong (in fact depending on the version of gcc there is a massive amount of warnings on unkown flags) -- put this is a pre-existing issue
- [x] ~~the structure of the project has to be cleaned up~~
- [x] ~~the git history has to be cleaned up~~

@ryanrhymes can I get your feedback on this? Do you have suggestions for the linking? Do you know anybody else that could point us in the right direction? I would like to avoid having to call `ocamlmklib` by hand and having to build the stubs completely by hand if possible (somewhat like https://github.com/SylvainBoilard/GLFW-OCaml/blob/master/dune or https://github.com/Chris00/L-BFGS-ocaml/blob/master/src/Lbfgsb.3.0/dune)

### CODE REVIEW

**Comment by mseri** [comment]
One possibility could be doing more by hand, as done by https://github.com/avsm/ocaml-yaml/blob/master/vendor/dune but we still need to build the full Eigen in some way first

**Comment by mseri** [comment]
With e58809e owl finally links eigen without issues (now I only have to figure out how to get rid of all
```
Error: Error on dynamically loaded library: ./src/owl/dllowl_stubs.so: ./src/owl/dllowl_stubs.so: undefined symbol: LAPACKE_dlagge
```
and friends to be able to test it properly. I still would like to find  way to build the components in order without having to do it by hands

**Comment by mseri** [comment]
ping @ryanrhymes

**Comment by ryanrhymes** [comment]
Hi Marcello, soooooo sorry for the late reply, I was overwhelmed by work after holiday. All changes look good to me, I don't have better suggestions on linking except the one I used before, i.e. using a separate Makefile. I tried jbuilder a long time ago but didn't manage to make it work.

I've included @mor1 in this issue, he knows better than me on building, so he can probably give more valuable comments?

File: eigen_cpp/lib/dune [hunk 1]
```diff
@@ -0,0 +1,16 @@
+(ignored_subdirs
+ (Eigen unsupported)
+)
+
+(rule
+ (targets libeigen.a libeigen.so)
+ (deps
+  (source_tree lib)
```
Comment (pveber) [inline eigen_cpp/lib/dune]: I think this line is incorrect (and the compilation fails for me here). Shouldn't it be: `(source_tree .)`?

**Comment by mor1** [comment]
Looking at this now. Sorry for the delay. Manic at the moment.

File: eigen_cpp/lib/dune (hunk 1 above)
Comment (mor1) [reply eigen_cpp/lib/dune]: Yes, it should :)

**Comment by mor1** [comment]
I created a separate PR (#10) with the fix indicated by @pveber thinking I could merge that, but I don't have maintainer privs on this repo.

@liang With that fix, looks ready to merge to me! Thanks @mseri @pveber !

File: eigen_cpp/lib/dune (hunk 1 above)
Comment (mseri) [reply eigen_cpp/lib/dune]: Ops, I forgot to commit the fix, sorry

**Comment by mseri** [comment]
I close this, as your pr contains the fixed version

**Pull request closed without merging**